packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -142,15 +141,14 @@ class AndroidSdk {
   }
 
   void _init() {
-    List<String> platforms = <String>[]; // android-22, ...
+    Iterable<Directory> platforms = <Directory>[]; // android-22, ...
 
     final Directory platformsDir = fs.directory(fs.path.join(directory, 'platforms'));
     if (platformsDir.existsSync()) {
       platforms = platformsDir
         .listSync()
-        .map<String>((FileSystemEntity entity) => fs.path.basename(entity.path))
-        .where((String name) => name.startsWith('android-'))
-        .toList();
+        .where((FileSystemEntity entity) => entity is Directory)
+        .map<Directory>((FileSystemEntity e) {final Directory d = e; return d;});
```

Comment:
This should be broken onto a couple lines... even though I admit I kind of like this.